pandas/tests/indexes/test_base.py

```diff
@@ -766,6 +766,30 @@ def test_sub(self):
         self.assertRaises(TypeError, lambda: idx - idx.tolist())
         self.assertRaises(TypeError, lambda: idx.tolist() - idx)
 
+    def test_map_identity_mapping(self):
+        # GH 12766
+        for name, cur_index in self.indices.items():
+            self.assert_index_equal(cur_index, cur_index.map(lambda x: x))
+
+    def test_map_that_returns_tuples_creates_multi_index(self):
+        # GH 12766
+        boolean_index = tm.makeIntIndex(3).map(lambda x: (x, x == 1))
+        expected = MultiIndex.from_tuples([(0, False), (1, True), (2, False)])
+        self.assert_index_equal(boolean_index, expected)
+
+    def test_map_that_returns_a_length_one_tuple_creates_an_index(self):
+        # GH 12766
+        boolean_index = tm.makeIntIndex(3).map(lambda x: (x, ))
+        expected = Index([(0, ), (1, ), (2, )])
+        self.assert_index_equal(boolean_index, expected)
+
+    def test_map_that_reduces_multi_index_to_single_index_returns_index(self):
```

Comment:
Can you shorten the test names a bit? 
(or else just include them in one single test, then you can add the long name/explanation as a comment)